Algorithm.CSharp/RegressionTests/Universes/ETFConstituentUniverseCompositeDelistingRegressionAlgorithm.cs

```diff
@@ -0,0 +1,173 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Interfaces;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Tests the delisting of the composite Symbol (ETF symbol) and the removal of
+    /// the universe and the symbol from the algorithm.
+    /// </summary>
+    public class ETFConstituentUniverseCompositeDelistingRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private Symbol _gdvd;
+        private Symbol _aapl;
+        private DateTime _delistingDate;
+        private int _universeSymbolCount;
+        private bool _universeAdded;
+        private bool _universeRemoved;
+        
+        /// <summary>
+        /// Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2020, 12, 1);
+            SetEndDate(2021, 1, 31);
+            SetCash(100000);
+            
+            UniverseSettings.Resolution = Resolution.Hour;
+            
+            _delistingDate = new DateTime(2021, 1, 21);
+
+            _aapl = AddEquity("AAPL", Resolution.Hour).Symbol;
+            _gdvd = AddEquity("GDVD", Resolution.Hour).Symbol;
+            
+            AddUniverse(new ETFConstituentsUniverse(_gdvd, UniverseSettings, FilterETFs));
```

Comment:
Adding `"GDVD"` as an equity isn't required right? Suggest adding a test for it too making sure everything works as expected 👍